src/components/storage-tracker/StorageTracker.tsx

```diff
@@ -0,0 +1,197 @@
+import { useEffect, useState } from "react";
+import { HardDrive, Infinity as InfinityIcon } from "@phosphor-icons/react";
+import { API_CONFIG } from "../../shared";
+
+interface StorageUsage {
+  username: string;
+  totalBytes: number;
+  fileCount: number;
+  isAdmin: boolean;
+  limitBytes: number;
+  remainingBytes: number;
+  usagePercentage: number;
+}
+
+interface StorageTrackerProps {
+  className?: string;
+}
+
+export function StorageTracker({ className = "" }: StorageTrackerProps) {
+  const [usage, setUsage] = useState<StorageUsage | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchStorageUsage = async () => {
+      try {
+        setIsLoading(true);
+        setError(null);
+
+        const jwt = localStorage.getItem("loresmith-jwt");
+        if (!jwt) {
+          setError("Not authenticated");
+          return;
+        }
+
+        const response = await fetch(
+          `${API_CONFIG.getApiBaseUrl()}/library/storage-usage`,
+          {
+            headers: {
+              Authorization: `Bearer ${jwt}`,
+            },
+          }
+        );
+
+        if (response.ok) {
+          const data = (await response.json()) as { usage: StorageUsage };
+          setUsage(data.usage);
+        } else {
+          setError("Failed to fetch storage usage");
+        }
+      } catch (err) {
+        setError("Error loading storage info");
+        console.error("Error fetching storage usage:", err);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchStorageUsage();
+
+    // Refresh every 30 seconds
+    const interval = setInterval(fetchStorageUsage, 30000);
+    return () => clearInterval(interval);
+  }, []);
+
+  const formatBytes = (bytes: number): string => {
+    if (bytes === 0) return "0 B";
+    if (bytes === Infinity) return "∞";
+    const k = 1024;
+    const sizes = ["B", "KB", "MB", "GB"];
+    const i = Math.floor(Math.log(bytes) / Math.log(k));
+    return `${parseFloat((bytes / k ** i).toFixed(2))} ${sizes[i]}`;
+  };
+
+  const getUsageColor = (percentage: number): string => {
+    if (percentage >= 90) return "text-red-600";
+    if (percentage >= 75) return "text-orange-600";
+    if (percentage >= 50) return "text-yellow-600";
+    return "text-green-600";
+  };
+
+  if (isLoading) {
+    return (
+      <div
+        className={`p-4 border-t border-neutral-200 dark:border-neutral-700 ${className}`}
```

Comment:
not to both our selves - we (i, this is strictly UI sadly), need to really reign in the one-off tailwind all over the place